cpp/daal/src/algorithms/low_order_moments/oneapi/low_order_moments_distributed_oneapi_impl.i

```diff
@@ -0,0 +1,488 @@
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesMeanVariance>::kFinalizeName = "finalizeMeanVariance";
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesMeanVariance>::kBldOptFNameSuff = " -DFNAMESUFF=MeanVariance ";
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesMeanVariance>::kBldOptScope = " -D_RMEAN_ -D_RVARC_ ";
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesMeanVariance>::kCacheKey =
+    "__daal_algorithms_low_order_moments_distributed_kernels_mean_variance";
+
+/* All task info estimatesAll parameters definitions */
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesAll>::kMergeDistrBlocksName = "mergeDistrBlocksAll";
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesAll>::kFinalizeName = "finalizeAll";
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesAll>::kBldOptFNameSuff = " -DFNAMESUFF=All ";
+
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesAll>::kBldOptScope =
+    " -D_RMIN_ -D_RMAX_ -D_RSUM_ -D_RSUM2_ -D_RSUM2C_ -D_RMEAN_ -D_RSORM_ -D_RVARC_ -D_RSTDEV_ -D_RVART_ ";
+template <>
+const char * TaskInfoDistributed<DAAL_FPTYPE, estimatesAll>::kCacheKey = "__daal_algorithms_low_order_moments_distributed_kernels_all";
+
+/*
+   Kernel methods implementation
+*/
+
+template <typename algorithmFPType, Method method>
+services::Status LowOrderMomentsDistributedKernelOneAPI<algorithmFPType, method>::compute(data_management::DataCollection * partialResultsCollection,
+                                                                                          PartialResult * partialResult, const Parameter * parameter)
+{
+    services::Status status;
+
+    auto & context = daal::services::internal::getDefaultContext();
+
+    if (method == defaultDense)
+    {
+        if (parameter->estimatesToCompute == estimatesMinMax)
+        {
+            LowOrderMomentsDistributedTaskOneAPI<algorithmFPType, estimatesMinMax> task(context, partialResultsCollection, partialResult, status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+        else if (parameter->estimatesToCompute == estimatesMeanVariance)
+        {
+            LowOrderMomentsDistributedTaskOneAPI<algorithmFPType, estimatesMeanVariance> task(context, partialResultsCollection, partialResult,
+                                                                                              status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+        else
+        {
+            /* estimatesAll */
+            LowOrderMomentsDistributedTaskOneAPI<algorithmFPType, estimatesAll> task(context, partialResultsCollection, partialResult, status);
+            DAAL_CHECK_STATUS_VAR(status);
+            return task.compute();
+        }
+    }
+
```

Comment:
Shouldn't we raise an `Not Implemented` error in other case?

Reply:
Thanks, added it.